pkg/pxc/app/statefulset/haproxy.go

```diff
@@ -207,6 +207,10 @@ func (c *HAProxy) AppContainer(spec *api.PodSpec, secrets string, cr *api.Percon
 		})
 	}
 
+	if cr.Spec.HAProxy.Lifecycle.PostStart != nil || cr.Spec.HAProxy.Lifecycle.PreStop != nil {
+		appc.Lifecycle = &cr.Spec.HAProxy.Lifecycle
+	}
+
```

Comment:
I believe we should check first that haproxy enabled in general. The same for proxysql

Reply:
My understanding is that this function will only be called if the haproxy is enabled in the cr, no? (same for proxysql)

Reply:
@nmarukovich do you think we still need to add a check if haproxy/proxysql is enabled?

Reply:
@nmarukovich I just wanted to check if you have reviewed this. Do we really need to add those checks? The function doesn't seem to be called if haproxy/proxysql are not enabled.

Reply:
Hey @ctria , I don't think you should check if it is enabled, but you should check if `cr.Spec.HAProxy` is not nil.